bot.go

```diff
@@ -57,15 +60,50 @@ func handleDownload(b *bot.Bot, m *bot.Message) {
 }
 
 func handleSearch(b *bot.Bot, m *bot.Message, api *rarbg.API) {
-	re, _ := regexp.Compile("tt[0-9]{7}") // e.g. tt0137523
-	imdbIDs := re.FindAllString(m.Text, -1)
+	imdbIDs, err := searchIMDbIDsFromMessage(m.Text)
+	if err != nil {
+		b.Send(m.Sender, replyNoIMDbIDS+err.Error())
+		return
+	}
 	if len(imdbIDs) == 0 {
 		b.Send(m.Sender, helpText)
 		return
 	}
+
 	for _, id := range imdbIDs {
 		result := new(bytes.Buffer)
 		searchIMDb(result, id, api)
 		b.Send(m.Sender, result.String(), &bot.SendOptions{ParseMode: bot.ModeMarkdown})
 	}
 }
+
+func searchIMDbIDsFromMessage(text string) ([]string, error) {
+	imdbIDs := []string{}
+	// Douban
+	movieLinks := findDoubanMovieURLs(text)
+	for _, url := range movieLinks {
+		movie := douban.NewMovie()
+		if err := movie.FetchFromURL(url); err != nil {
+			return nil, err
+		}
+		imdbIDs = append(imdbIDs, movie.IMDbID())
+	}
+	// IMDB
+	if len(imdbIDs) == 0 {
```

Comment:
Maybe we can support messages that contained links of both Douban and IMDb.

Reply:
It's a good idea to support multiple kinds of links in one message, but I suggest to do a refactoring before we do more.

And I also worried that If we don't introduce some async model, doing more things in the message handler will be dangerous.

Reply:
I just thought to have this done with a simple change.

I’m not sure about how many links will be in a message, but refactoring is good.

Reply:
![screen shot 2018-01-07 at 6 54 24 pm](https://user-images.githubusercontent.com/408385/34648800-35590d78-f3dc-11e7-8c8b-e21440efec95.png)

Reply:
I see, let's do this!

Could you edit this PR?